Framework/PythonInterface/plugins/algorithms/SaveGEMMAUDParamFile.py

```diff
@@ -0,0 +1,204 @@
+        focused spectrum i with
+        :return: a list of m values, where each value is the assigned parameter for the
+        corresponding bank
+        """
+        return (bank_param_list[grouping_scheme[spec_num] - 1] for spec_num in spectrum_numbers)
+
+    def _find_isis_powder_dir(self):
+        script_dirs = config["pythonscripts.directories"].split(";")
+        diffraction_dir = None
+        for dir in script_dirs:
+            if "Diffraction" in dir:
+                diffraction_dir = dir
+
+        if diffraction_dir is None:
+            logger.warning("Could not find default diffraction directory for .maud template file: "
+                           "you'll have to find it yourself")
+            return ""
+        else:
+            return os.path.join(diffraction_dir, "isis_powder", "gem_routines", "maud_param_template.maud")
+
+    def _format_param_list(self, param_list):
+        return "\n".join(str(param) for param in param_list)
+
+    def _get_two_theta_and_phi(self, bank):
+        instrument = bank.getInstrument()
+        detector = bank.getDetector(0)
+
+        sample_pos = instrument.getSample().getPos()
+        source_pos = instrument.getSource().getPos()
+        det_pos = detector.getPos()
+
+        beam_dir = sample_pos - source_pos
+        detector_dir = det_pos - sample_pos
+
+        return math.degrees(beam_dir.angle(detector_dir)), math.degrees(detector.getPhi())
+
+    def _parse_gsas_param_file(self, gsas_filename):
+        with open(gsas_filename) as prm_file:
+            gsas_params_lines = prm_file.read().split("\n")
+
+        distances = []
+        difcs = []
+        difas = []
+        tzeros = []
+        alpha_zeros = []
+        alpha_ones = []
+        beta_zeros = []
+        beta_ones = []
+        sigma_zeros = []
+        sigma_ones = []
+        sigma_twos = []
+        for line in gsas_params_lines:
+            line_items = line.split()
+            if re.match(self.BANK_PARAMS_LINE, line):
+                distances.append(float(line_items[2]))
+            elif re.match(self.DIFFRACTOMETER_CONSTANTS_LINE, line):
+                difcs.append(float(line_items[3]))
+                difas.append(float(line_items[4]))
+                tzeros.append(float(line_items[5]))
+            elif re.match(self.PROFILE_COEFFS_LINE_1, line):
+                alpha_zeros.append(float(line_items[3]))
+                alpha_ones.append(float(line_items[4]))
+                beta_zeros.append(float(line_items[5]))
+                beta_ones.append(float(line_items[6]))
+            elif re.match(self.PROFILE_COEFFS_LINE_2, line):
+                sigma_zeros.append(float(line_items[3]))
+                sigma_ones.append(float(line_items[4]))
+                sigma_twos.append(float(line_items[5]))
+
+        return distances, difas, difcs, tzeros, alpha_zeros, alpha_ones, \
+            beta_zeros, beta_ones, sigma_zeros, sigma_ones, sigma_twos
```

Comment:
Could we maybe tidy up some of this by sticking all the entries in a `dict` instead of a tuple? It feels like the more natural data structure.

I don't think ordering is particular important here, but we could use an ordered dict if that is necessary.  You can also use a default dict to initialise the entires to a list by default instead of checking where a key exists yet.